src/test/java/de/hpi/bpt/argos/util/ArgosTestUtil.java

```diff
@@ -0,0 +1,242 @@
+package de.hpi.bpt.argos.util;
+
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.attribute.Attribute;
+import de.hpi.bpt.argos.storage.dataModel.attribute.AttributeImpl;
+import de.hpi.bpt.argos.storage.dataModel.attribute.type.TypeAttribute;
+import de.hpi.bpt.argos.storage.dataModel.attribute.type.TypeAttributeImpl;
+import de.hpi.bpt.argos.storage.dataModel.entity.Entity;
+import de.hpi.bpt.argos.storage.dataModel.entity.EntityImpl;
+import de.hpi.bpt.argos.storage.dataModel.entity.type.EntityType;
+import de.hpi.bpt.argos.storage.dataModel.entity.type.EntityTypeImpl;
+import de.hpi.bpt.argos.storage.dataModel.event.Event;
+import de.hpi.bpt.argos.storage.dataModel.event.EventImpl;
+import de.hpi.bpt.argos.storage.dataModel.event.type.EventType;
+import de.hpi.bpt.argos.storage.dataModel.event.type.EventTypeImpl;
+import de.hpi.bpt.argos.storage.dataModel.mapping.EventEntityMapping;
+import de.hpi.bpt.argos.storage.dataModel.mapping.EventEntityMappingImpl;
+import de.hpi.bpt.argos.storage.dataModel.mapping.MappingCondition;
+import de.hpi.bpt.argos.storage.dataModel.mapping.MappingConditionImpl;
+import javafx.util.Pair;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+public class ArgosTestUtil {
+	private static final Random random = new Random();
+
+	public static String getCurrentTimestamp() {
+		return new SimpleDateFormat("dd_MM_yyyy_HH_mm_ss").format(new Date());
+	}
+
+	public static int getRandomInteger(int min, int max) {
+		return random.nextInt(max) + min;
+	}
+
+	public static long getRandomLong() {
+		return random.nextLong();
+	}
+
+	public static float getRandomFloat() {
+		return random.nextFloat();
+	}
+
+	public static String getRandomString() {
+		return UUID.randomUUID().toString();
+	}
+
+	public static EntityType createEntityType(boolean saveInDatabase) {
+		EntityType newEntityType = new EntityTypeImpl();
+
+		newEntityType.setName("EntityType_" + getCurrentTimestamp());
+		newEntityType.setParentId(-1); // directly under root node
+
+		if (saveInDatabase) {
+			PersistenceAdapterImpl.getInstance().saveArtifacts(newEntityType);
+		}
+
+		return newEntityType;
+	}
+
+	public static EntityType createEntityType(EntityType parent, boolean saveInDatabase) {
+		EntityType newEntityType = createEntityType(false);
+
+		newEntityType.setParentId(parent.getId());
+
+		if (saveInDatabase) {
+			PersistenceAdapterImpl.getInstance().saveArtifacts(newEntityType);
+		}
+
+		return newEntityType;
+	}
+
+	public static List<TypeAttribute> createEntityTypeAttributes(EntityType owner, boolean saveInDatabase) {
+		return createTypeAttributes(owner.getId(), saveInDatabase);
+	}
+
+	public static Entity createEntity(EntityType type, boolean saveInDatabase) {
+		Entity newEntity = new EntityImpl();
+
+		newEntity.setName("Entity_" + getCurrentTimestamp());
```

Comment:
This is to make sure this `EntityType` does not exist in the database whatsoever. Since the timestamp is truely unique I considered it to be more safe than just a random string.